python/lsst/ts/scriptqueue/ui/request_cmd.py

```diff
@@ -0,0 +1,385 @@
+        except Exception as e:
+            self.log.exception(e)
+
+        return False
+
+    def do_enable(self, arg):
+        """Enable the queue."""
+        try:
+            self.model.enable_queue()
+        except Exception as e:
+            self.log.exception(e)
+
+        return False
+
+    def do_list(self, arg):
+        """List available scripts."""
+        try:
+            print("Listing all available scripts.")
+
+            script_list = self.model.list()
+
+            print('List of external scripts:')
+            for script in script_list["external"]:
+                print(f'\t - {script}', )
+
+            print('List of standard scripts:')
+            for script in script_list["standard"]:
+                print(f'\t - {script}', )
+        except Exception as e:
+            self.log.exception(e)
+
+        return False
+
+    def do_pause(self, arg):
+        """Pause the queue."""
+        print("Pausing queue.")
+        try:
+            self.model.pause_queue()
+        except Exception as e:
+            self.log.exception(e)
+
+        return False
+
+    def do_resume(self, arg):
+        """Resume queue."""
+        print("Resuming queue.")
+        try:
+            self.model.resume_queue()
+        except Exception as e:
+            self.log.exception(e)
+
+    def do_exit(self, arg):
+        """Exit shell."""
+        print("Exiting...")
+        return True
+
+    def do_quitQueue(self, arg):
+        """Interrupt the queue by sending it the exitControl command and exit request script.
+        If it fails to quit the queue
+        """
+        try:
+            retval = self.model.quit_queue()
+            if retval:
+                print("Bye...")
+            else:
+                print("Could not quit queue.")
+        except Exception as e:
+            print("Could not quit queue.")
+            self.log.exception(e)
+            retval = False
+
+        return retval
+
+    def do_stop(self, args):
+        """Stop list of scripts.
+
+        Parameters
+        ----------
+        args : str
+            Comma separated list of scripts indices.
+        """
+        print(f'Stopping script {args}.')
+        try:
+            sal_indices = [int(i) for i in args.split(",")]
+            self.model.stop_scripts(sal_indices, terminate=False)
+        except Exception as e:
+            self.log.exception(e)
+
+        return 0
+
+    def do_terminate(self, args):
+        """Terminate list of scripts. Similar to stop but will set `terminate` flag to True.
+
+        Parameters
+        ----------
+        args : str
+            Comma separated list of scripts indices.
+        """
+        print(f'Terminating script {args}')
+        try:
+            sal_indices = [int(i) for i in args.split(",")]
+            self.model.stop_scripts(sal_indices, terminate=True)
+        except Exception as e:
+            self.log.exception(e)
+
+        return 0
+
+    def do_show(self, args):
```

Comment:
Why is this a command? I would naively expect the user to always want to see the state of the queue. Though I suppose it could get ugly if the user is trying to type something while all this useful data is displayed. I fear `curses` might be a more sensible UI, though it looks a nuisance to program.

Reply:
Yes, I though about using `curses` but I fear it would be a much larger effort to get something nice and that eventually would be replaced by the LOVE interface. Using `cmd` has the drawback that I can't use the callback methods because the library is not `asyncio` friendly. I spent a good chunk of time trying to get them to play nicely but I couldn't. 

I left some of the callback methods there so we can evaluate later if doing a curses version of it would worth it but I honestly doubt it.

Reply:
I agree